app/models/health/agency_performance.rb

```diff
@@ -118,243 +94,101 @@ def patient_referrals
-        warehouse_dates = warehouse_ssm_dates_by_patient_id
-        epic_dates = epic_ssms_by_patient_id
-
-        # determine most recent within range for each patient
-        patients_with_dates_from_various_sources([hmis_dates, warehouse_dates, epic_dates])
-      end
-    end
-
-    def warehouse_ssm_dates_by_patient_id
-      @warehouse_ssm_dates_by_patient_id ||= Health::SelfSufficiencyMatrixForm.completed.
-        distinct.
-        where(patient_id: patient_referrals.keys). # limit to patients in scope
-        order(completed_at: :asc).
-        pluck(:patient_id, :completed_at).to_h
-    end
-
-    def hmis_ssms_max_dates_by_patient_id
-      @hmis_ssms_max_dates_by_patient_id ||=
-        hmis_ssms_max_dates_by_client_id.map do |client_id, date|
-          [@client_ids[client_id], date]
-        end.to_h
-    end
-
-    def hmis_ssms_max_dates_by_client_id
-      @hmis_ssms_max_dates_by_client_id ||= GrdaWarehouse::Hud::Client.joins(:source_hmis_forms).
-        merge(GrdaWarehouse::HmisForm.self_sufficiency).
-        distinct.
-        where(id: client_ids.keys). # limit to clients in scope
-        pluck(:id, hmis_form_t[:collected_at].to_sql).
-        to_h
-    end
-
-    def epic_ssms_by_patient_id
-      @epic_ssms_by_patient_id ||= Health::EpicSsm.distinct.
-        joins(:patient).
-        where(patient_id: patient_referrals.keys). # limit to patients in scope
-        order(ssm_updated_at: :asc).
-        pluck(hp_t[:id].to_sql, :ssm_updated_at).to_h
-    end
-
-    # CHAS
-    def cha_dates
-      @cha_dates ||= begin
-        warehouse_dates = warehouse_cha_dates_by_patient_id
-        epic_dates = epic_chas_by_patient_id
-
-        # determine most recent within range for each patient
-        patients_with_dates_from_various_sources([warehouse_dates, epic_dates])
-      end
-    end
-
-    def warehouse_cha_dates_by_patient_id
-      @warehouse_cha_dates_by_patient_id ||= Health::ComprehensiveHealthAssessment.completed.
-        distinct.
-        where(patient_id: patient_referrals.keys). # limit to patients in scope
-        order(completed_at: :asc).
-        pluck(:patient_id, :completed_at).to_h
-    end
-
-    def epic_chas_by_patient_id
-      @epic_chas_by_patient_id ||= Health::EpicCha.distinct.
-        joins(:patient).
-        where(hp_t[:id].in(patient_referrals.keys)). # limit to patients in scope
-        order(cha_updated_at: :asc).
-        pluck(hp_t[:id].to_sql, :cha_updated_at).to_h
-    end
-
-    # Careplans
-    def careplan_dates
-      @careplan_dates ||= begin
-        warehouse_dates = warehouse_careplan_dates_by_patient_id
-        # Ignore Epic care plans for now, 8/31/2018
-        # epic_dates = epic_careplan_by_patient_id
-        epic_dates = {}
-        # determine most recent within range for each patient
-        patients_with_dates_from_various_sources([warehouse_dates, epic_dates])
-      end
+    def patient_ids
+      @patient_ids ||= patient_referrals.keys
```

Comment:
Does this need any "active" filter?

Reply:
`patient_referrals` uses `active_within_range`, which should cover it.